### PR TITLE
cpu/esp32: add support for ESP32-C3 SoC family

### PR DESCRIPTION
**Pull request by gschorcht**
### Contribution description

This PR provides the support for the ESP32-C3 SoC family and the ESP32-C3-DevKit board.

For the moment, this PR includes ~PR #17601,~ PR #17841 and PR #17842 to be compilable. Once these PRs are merged this PR is rebased. The first additional commit of this PR is d0210bea00954d2ba8bf07c1b980f5461ee11a2a.

### Testing procedure

Compilation in CI has to succeed. All peripheral and basis tests have to pass.

### Issues/PRs references

~Depends on PR #17601~
Depends on PR #17841
Depends on PR #17842
Prerequisite for PR #18185

### CODE REVIEW

**Comment by gschorcht** [comment]
Let's hope this PR and the master don't diverge so much until these PRs can be merged. Any required rebase after other changes to `cpu/esp*` will cause extensive conflicts due to the fundamental changes regarding the basic structure of the implementation and how the ESP IDF is used now. Preferably, I wouldn't allow any other changes to `cpu/esp*` anymore until these PRs are merged.

File: cpu/esp32/Makefile.features [hunk 1]
```diff
@@ -1,14 +1,25 @@
-CPU_FAM = esp32
+ifneq (,$(filter esp32,$(CPU_FAM)))
+  CPU_ARCH = xtensa
+  CPU_CORE = xtensa-lx6
+else ifneq (,$(filter esp32c3,$(CPU_FAM)))
+  CPU_ARCH = riscv_esp32
+  CPU_CORE = riscv32
```
Comment (kfessel) [inline cpu/esp32/Makefile.features:6]: There are allready some riscv32 MCU they use 
`CPU_CORE := rv32imac` (eg.: cpu/gd32v/Makefile.features)
is there a reason to have this `riscv32`

Comment (gschorcht) [reply cpu/esp32/Makefile.features:6]: Unfortunately, we can't use `cpu/riscv_common` as is, please see section "Background" in PR https://github.com/RIOT-OS/RIOT/pull/18260#issue-1284563955. In short
- ESP-IDF's `vector.S` and `portasm.S`, which we use, use a different context frame structure
- RISC-V based ESP32x SoCs use a special SoC register ` INTERRUPT_CORE0_CPU_INT_THRESH_REG` to enable/disable interrupts.

Comment (gschorcht) [reply cpu/esp32/Makefile.features:6]: Ah ok, now I got it. I must have read too superficially. You don't ask why not use `cpu/esp_common`, but why not use `rv32` as `CPU_CORE`? To be honest, I have to check whether this makes a difference.

**Comment by gschorcht** [comment]
All split-offs are merged now.

**Pull request closed without merging**